addons/binding/org.openhab.binding.tankerkoenig/src/main/java/org/openhab/binding/tankerkoenig/internal/serializer/CustomTankerkoenigListResultDeserializer.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.openhab.binding.tankerkoenig.internal.serializer;
+
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import org.openhab.binding.tankerkoenig.internal.config.LittleStation;
+import org.openhab.binding.tankerkoenig.internal.config.Prices;
+import org.openhab.binding.tankerkoenig.internal.config.TankerkoenigListResult;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+
+/***
+ * Custom Deserializer fopr the list result of tankerkoenigs api response
+ *
+ * @author Dennis Dollinger
+ *
+ */
+public class CustomTankerkoenigListResultDeserializer implements JsonDeserializer<TankerkoenigListResult> {
+
+    @Override
+    public TankerkoenigListResult deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context)
+            throws JsonParseException {
+
+        final JsonObject jsonObject = json.getAsJsonObject();
+        final Gson gson = new Gson();
```

Comment:
This must not be instantiated on every method execution - please use a field for it instead.